### PR TITLE
Settings

### PR DESCRIPTION
**Pull request by Dagmawit-AHG**

### CODE REVIEW

File: TrainingApp/AppDelegate.swift [hunk 1]
```diff
@@ -19,6 +19,26 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                UIApplication.shared.windows.first?.overrideUserInterfaceStyle = .dark
+            }
+        }
+        
+        if UserDefaults.standard.object(forKey: R.string.localizable.language()) != nil && UserDefaults.standard.object(forKey: R.string.localizable.language()) as! String == "pl" {
+            UserDefaults.standard.set("pl",forKey: R.string.localizable.language())
+//            Bundle.setLanguage((targetLang != nil) ? targetLang! : "en")
+        }
+//        else if UserDefaults.standard.object(forKey: "Language") != nil && UserDefaults.standard.object(forKey: "Language") as! String == "en" {
+//            UserDefaults.standard.set("en", forKey: "Language")
```
Comment (vladislavmatygin) [inline TrainingApp/AppDelegate.swift]: remove comments

File: TrainingApp/Controller/Authentication/LoginViewController.swift [hunk 2]
```diff
@@ -47,6 +48,7 @@ final class LoginViewController: UIViewController {
                 alert.addAction(UIAlertAction(title: R.string.localizable.oK(), style: UIAlertAction.Style.default, handler: nil))
                 self?.present(alert, animated: true, completion: nil)
             }
+            self?.userDefaults.set(true, forKey: R.string.localizable.isuserloggediN())
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/Authentication/LoginViewController.swift]: change naming "isuserloggediN"

File: TrainingApp/Controller/LanguageViewController.swift [hunk 3]
```diff
@@ -128,15 +131,33 @@ final class LanguageViewController: UIViewController, UITableViewDataSource, UIT
...
             englishUKCheck.isHidden = true
+            Bundle.setLanguage("en")
+            UserDefaults.standard.set("en",forKey: "Language")
+            languageLabel.text = R.string.localizable.language().localizableString("en")
+            suggestedLangLabel.text = R.string.localizable.suggestedLanguages().localizableString("en")
+            englishUSLabel.text = R.string.localizable.englishUS().localizableString("en")
+            polishLabel.text = R.string.localizable.polish().localizableString("en")
+            englishUKLabel.text = R.string.localizable.englishUK().localizableString("en")
+            longTextLabel.text = R.string.localizable.appWillUse().localizableString("en")
+            otherLangLabel.text = R.string.localizable.otherLanguages().localizableString("en")
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/LanguageViewController.swift:147]: i think it is not the best way of using different languages in application (let's get on call and discuss it)

File: TrainingApp/Controller/SettingsViewController.swift [hunk 4]
```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
...
+        
+        discountDealsSwitch.setOnValueChangeListener {
+            if self.discountDealsSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_DISCOUNT)
+                self.userDefaults.set(self.DISCOUNT_ON, forKey: self.DISCOUNT_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_DISCOUNT)
+                self.userDefaults.set(self.DISCOUNT_OFF, forKey: self.DISCOUNT_KEY)
+            }
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift:170]: Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

File: TrainingApp/Controller/SettingsViewController.swift [hunk 5]
```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
...
+        
+        executiveProgramSwitch.setOnValueChangeListener {
+            if self.executiveProgramSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_EXECUTIVE)
+                self.userDefaults.set(self.EXECUTIVE_ON, forKey: self.EXECUTIVE_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_EXECUTIVE)
+                self.userDefaults.set(self.EXECUTIVE_OFF, forKey: self.EXECUTIVE_KEY)
+            }
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift:159]: Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

File: TrainingApp/Controller/SettingsViewController.swift [hunk 6]
```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
...
+        
+        flightUpdatesSwitch.setOnValueChangeListener {
+            if self.flightUpdatesSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_FLIGHT)
+                self.userDefaults.set(self.FLIGHT_ON, forKey: self.FLIGHT_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_FLIGHT)
+                self.userDefaults.set(self.FLIGHT_OFF, forKey: self.FLIGHT_KEY)
+            }
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift:148]: Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

File: TrainingApp/Controller/SettingsViewController.swift [hunk 7]
```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
...
+        notificationsSwitch.setOnValueChangeListener {
+            if self.notificationsSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_NOTIFICATION)
+                self.userDefaults.set(self.NOTIFICATION_ON, forKey: self.NOTIFICATION_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_NOTIFICATION)
+                self.userDefaults.set(self.NOTIFICATION_OFF, forKey: self.NOTIFICATION_KEY)
+            }
+            self.updateNotifications()
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift:137]: Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

File: TrainingApp/Controller/SettingsViewController.swift [hunk 8]
```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
...
+        darkModeSwitch.setOnValueChangeListener {
+            if self.darkModeSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_THEME)
+                self.userDefaults.set(self.DARK_THEME, forKey: self.THEME_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_THEME)
+                self.userDefaults.set(self.LIGHT_THEME, forKey: self.THEME_KEY)
+            }
+            self.updateTheme()
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift:125]: Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

File: TrainingApp/Controller/SettingsViewController.swift [hunk 9]
```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
...
-    @IBAction private func darkModeTogglePressed(_ sender: UIButton) {
-        if darkModeToggleButton.currentImage == R.image.button() {
-            darkModeIsOn()
-        }
-        else {
-            darkModeIsOff()
+        checkThemeSwitchState()
+        checkNotificationSwitchState()
+        print(FLIGHT_KEY)
+        print(EXECUTIVE_KEY)
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove prints

File: TrainingApp/Controller/SettingsViewController.swift [hunk 10]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
     
     // MARK: - Helpers
     
+    private func checkThemeSwitchState() {
+        if(userDefaults.bool(forKey: ON_OFF_KEY_THEME)) {
+            darkModeSwitch.setOn(true, animated: false)
+            self.userDefaults.set(self.DARK_THEME, forKey: self.THEME_KEY)
+        }
+        else {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 11]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
...
+    
+    private func checkNotificationSwitchState() {
+        if(userDefaults.bool(forKey: ON_OFF_KEY_NOTIFICATION)) {
+            notificationsSwitch.setOn(true, animated: false)
+            self.userDefaults.set(self.NOTIFICATION_ON, forKey: self.NOTIFICATION_KEY)
+            checkFlightSwitchState()
+            checkExecutiveSwitchState()
+            checkDiscountSwitchState()
+        }
+        else {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 12]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
...
+            checkDiscountSwitchState()
+        }
+    }
+    
+    private func checkFlightSwitchState() {
+        if(userDefaults.bool(forKey: ON_OFF_KEY_FLIGHT)) {
+            flightUpdatesSwitch.setOn(true, animated: false)
+            self.userDefaults.set(self.FLIGHT_ON, forKey: self.FLIGHT_KEY)
+        }
+        else {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 13]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
...
+            flightUpdatesSwitch.setOn(false, animated: false)
+            self.userDefaults.set(self.FLIGHT_OFF, forKey: self.FLIGHT_KEY)
+        }
+    }
+    
+    private func checkExecutiveSwitchState() {
+        if(userDefaults.bool(forKey: ON_OFF_KEY_EXECUTIVE)) {
+            executiveProgramSwitch.setOn(true, animated: false)
+            self.userDefaults.set(self.EXECUTIVE_ON, forKey: EXECUTIVE_KEY)
+        }
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 14]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
...
+            self.userDefaults.set(self.EXECUTIVE_OFF, forKey: EXECUTIVE_KEY)
+        }
+    }
+    
+    private func checkDiscountSwitchState() {
+        if(userDefaults.bool(forKey: ON_OFF_KEY_DISCOUNT)) {
+            discountDealsSwitch.setOn(true, animated: false)
+            self.userDefaults.set(self.DISCOUNT_ON, forKey: DISCOUNT_KEY)
+        }
+        else {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 15]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
...
+            discountDealsSwitch.setOn(false, animated: false)
+            self.userDefaults.set(self.DISCOUNT_OFF, forKey: DISCOUNT_KEY)
+        }
+    }
+    
+    private func updateTheme() {
+        let theme = userDefaults.string(forKey: THEME_KEY)
+        if (theme == LIGHT_THEME) {
+            darkModeIsOff()
+        }
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 16]
```diff
@@ -119,6 +174,87 @@ final class SettingsViewController: UIViewController {
...
+        else if (theme == DARK_THEME) {
+            darkModeIsOn()
+        }
+    }
+    
+    private func updateNotifications() {
+        let notifications = userDefaults.string(forKey: NOTIFICATION_KEY)
+        if (notifications == NOTIFICATION_ON) {
+            notificationIsOn()
+        }
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: remove this space
make it 
if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 17]
```diff
@@ -143,76 +279,40 @@ final class SettingsViewController: UIViewController {
...
         bigRectangle.image = R.image.rectangle_big()
         flightUpdatesLabel.isHidden = false
-        flightUpdatesToggle.isHidden = false
+        flightUpdatesSwitch.isHidden = false
         executiveProgramLabel.isHidden = false
-        executiveProgramToggle.isHidden = false
+        executiveProgramSwitch.isHidden = false
         discountDealsLabel.isHidden = false
-        discountDealsToggle.isHidden = false
+        discountDealsSwitch.isHidden = false
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: create function 
private func setupShowingLabels(isHidden: Bool) {
     flightUpdatesToggle.isHidden = isHidden
         flightUpdatesSwitch.isHidden = isHidden
         executiveProgramLabel.isHidden = isHidden
         executiveProgramToggle.isHidden = isHidden
         executiveProgramSwitch.isHidden = isHidden
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 18]
```diff
@@ -143,76 +279,40 @@ final class SettingsViewController: UIViewController {
...
     private func notificationIsOff() {
-        notificationsToggleButton.setImage(R.image.button(), for: .normal)
         notificationsRectangle.isHidden = false
         bigRectangle.isHidden = true
         flightUpdatesLabel.isHidden = true
-        flightUpdatesToggle.isHidden = true
+        flightUpdatesSwitch.isHidden = true
         executiveProgramLabel.isHidden = true
-        executiveProgramToggle.isHidden = true
+        executiveProgramSwitch.isHidden = true
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: create function 
private func setupShowingLabels(isHidden: Bool) {
     flightUpdatesToggle.isHidden = isHidden
         flightUpdatesSwitch.isHidden = isHidden
         executiveProgramLabel.isHidden = isHidden
         executiveProgramToggle.isHidden = isHidden
         executiveProgramSwitch.isHidden = isHidden
}

File: TrainingApp/SceneDelegate.swift [hunk 19]
```diff
@@ -18,8 +18,14 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         window = UIWindow(windowScene: scene)
         
         let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        var rootVC = UIViewController()
+        if(UserDefaults.standard.bool(forKey: R.string.localizable.isuserloggediN()) == true) {
+            rootVC = storyboard.instantiateViewController(withIdentifier: "HomeViewController")
+        }
+        else {
```
Comment (vladislavmatygin) [inline TrainingApp/SceneDelegate.swift]: remove this space
make it
if {
} else {
}

File: TrainingApp/Utils/AppHelper.swift [hunk 20]
```diff
@@ -0,0 +1,19 @@
+//
+//  AppHelper.swift
+//  TrainingApp
+//
+//  Created by Dagmawit Alemayehu on 24/01/2023.
+//
+
+import CoreData
```
Comment (vladislavmatygin) [inline TrainingApp/Utils/AppHelper.swift:8]: did you use CoreData? if not - remove it please

File: TrainingApp/Utils/AppHelper.swift [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+
+class AppHelper: NSManagedObject {
+    
+    public static func localized(_ key: String, language: String? = nil) -> String {
+            let language = language ?? Locale.preferredLanguages.first?.components(separatedBy: "-").first ?? "en"
+            guard let path = Bundle.main.path(forResource: language, ofType: "lproj"), let bundle = Bundle(path: path) else {
+                return Bundle.main.localizedString(forKey: key, value: nil, table: nil)
+            }
```
Comment (vladislavmatygin) [inline TrainingApp/Utils/AppHelper.swift]: Space after guard let

**Review by vladislavmatygin — COMMENTED** [review]
fix it please
If you need me, let's get on call

**Comment by vladislavmatygin** [comment]
For now let’s fix merge issues with Settings
and then let’s work like this (git flow: https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow)

File: TrainingApp/AppDelegate.swift (hunk 1 above)
Comment (Dagmawit-AHG) [reply TrainingApp/AppDelegate.swift]: Done 👍

File: TrainingApp/Controller/Authentication/LoginViewController.swift (hunk 2 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/Authentication/LoginViewController.swift]: Done 👍
I changed it to "userStatus"

File: TrainingApp/Controller/SettingsViewController.swift (hunk 4 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift:170]: Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.

File: TrainingApp/Controller/SettingsViewController.swift (hunk 5 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift:159]: Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.

File: TrainingApp/Controller/SettingsViewController.swift (hunk 6 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift:148]: Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.

File: TrainingApp/Controller/SettingsViewController.swift (hunk 7 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift:137]: Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.

File: TrainingApp/Controller/SettingsViewController.swift (hunk 8 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift:125]: Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.

File: TrainingApp/Controller/SettingsViewController.swift (hunk 9 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 10 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 11 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 12 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 13 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 15 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 16 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 17 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 18 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/SceneDelegate.swift (hunk 19 above)
Comment (Dagmawit-AHG) [reply TrainingApp/SceneDelegate.swift]: Done 👍

File: TrainingApp/Utils/AppHelper.swift (hunk 20 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Utils/AppHelper.swift:8]: Yes, I'm using it for 'NSManagedObject' which is inherited by the class.

File: TrainingApp/Utils/AppHelper.swift (hunk 21 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Utils/AppHelper.swift]: Done 👍

File: TrainingApp/Controller/LanguageViewController.swift (hunk 3 above)
Comment (vladislavmatygin) [reply TrainingApp/Controller/LanguageViewController.swift:147]: @Dagmawit-AHG what about it?

File: TrainingApp/Controller/LanguageViewController.swift [hunk 22]
```diff
@@ -66,22 +71,23 @@ final class LanguageViewController: UIViewController, UITableViewDataSource, UIT
...
-            let cell = tableView.dequeueReusableCell(withIdentifier: R.string.localizable.languageCell(), for: indexPath)
-            cell.textLabel?.text = Array(languages.keys)[indexPath.row]
-            cell.textLabel?.font.withSize(18.0)
-            print(Array(languages.keys)[indexPath.row])
+            var cell = UITableViewCell()
+                languagesTableView.backgroundView = UIImageView(image: R.image.rectangle_all())
+                cell = tableView.dequeueReusableCell(withIdentifier: R.string.localizable.languageCell(), for: indexPath)
+                cell.textLabel?.text = Array(languages.keys)[indexPath.row]
+                cell.textLabel?.font.withSize(18.0)
+                print(Array(languages.keys)[indexPath.row])
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/LanguageViewController.swift]: remove print()

File: TrainingApp/Controller/SettingsViewController.swift [hunk 23]
```diff
@@ -60,13 +60,62 @@ final class SettingsViewController: UIViewController {
...
         updateTheme()
         updateNotifications()
-
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        if self.darkModeSwitch.isOn {
+            self.darkModeIsOn()
+        }
+        else {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 24]
```diff
@@ -60,13 +60,62 @@ final class SettingsViewController: UIViewController {
...
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == R.string.localizable.signOutSegue() {
+            guard let destinationVC = segue.destination as? StartViewController else { return }
+            
+            destinationVC.modalPresentationStyle = .fullScreen
+            destinationVC.navigationController?.setNavigationBarHidden(false, animated: true)
+            present(destinationVC, animated: true, completion: nil)
+        }
+        else if segue.identifier == R.string.localizable.backToHomepageSegue() {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: if {
} else {
}

File: TrainingApp/Controller/SettingsViewController.swift [hunk 25]
```diff
@@ -60,13 +60,62 @@ final class SettingsViewController: UIViewController {
...
+            guard let destinationVC = segue.destination as? HomeViewController else { return }
+            
+            destinationVC.modalPresentationStyle = .fullScreen
+            destinationVC.navigationController?.setNavigationBarHidden(false, animated: true)
+            present(destinationVC, animated: true, completion: nil)
+        }
+        else if segue.identifier == R.string.localizable.showLanguagesPage() {
```
Comment (vladislavmatygin) [inline TrainingApp/Controller/SettingsViewController.swift]: if {
} else {
}

**Review by vladislavmatygin — COMMENTED** [review]
Let's fix this issues and finish this merge

File: TrainingApp/Controller/LanguageViewController.swift (hunk 3 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/LanguageViewController.swift:147]: I have removed it from the code. The labels can change in language but only if I leave the page and reopen it again.

File: TrainingApp/Controller/SettingsViewController.swift (hunk 14 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/LanguageViewController.swift (hunk 3 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/LanguageViewController.swift:147]: I pushed the changes to the homeVC branch because it's an issue of localization

Comment (vladislavmatygin) [reply TrainingApp/Controller/LanguageViewController.swift:147]: OK👌  Let's leave it to next PR

File: TrainingApp/Controller/LanguageViewController.swift (hunk 22 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/LanguageViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 23 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 24 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

File: TrainingApp/Controller/SettingsViewController.swift (hunk 25 above)
Comment (Dagmawit-AHG) [reply TrainingApp/Controller/SettingsViewController.swift]: Done 👍

**Comment by vladislavmatygin** [comment]
Everything looks fine, you can merge it 🥳

**Pull request merged**